overwatch-rs/src/services/relay.rs

```diff
@@ -157,7 +157,7 @@ impl<S: ServiceCore> Relay<S> {
 
     #[instrument(skip_all, err(Debug))]
     async fn handle_relay_response(
-        &mut self,
+        &self,
```

Comment:
Looks good, are there are any other `&mut` that should be just `&` ? If so please add to this pr. If not omit this :)